### ISSUE TITLE
How to override default coredns config

### ISSUE THREAD

**Issue by Vandersteen** [Azure/AKS#1111]
As the proxy plugin has been deprecated and is giving us issues, we would like to use the forward plugin instead. However I would need to be able to override the coredns config for this.

So:

From

```
.:53 {
    errors
    health
    kubernetes cluster.local in-addr.arpa ip6.arpa {
      pods insecure
      upstream
      fallthrough in-addr.arpa ip6.arpa
    }
    prometheus :9153
    proxy . /etc/resolv.conf
    cache 30
    loop
    reload
    loadbalance
    import custom/*.override
}
```

To 

```
.:53 {
    ...
    forward . /etc/resolv.conf
    ...
}
```

However, as noted in the documentation, we cannot edit this information, only 'extend' it.

Is it possible to change this ?
If so would this suffise ?

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns-custom
  namespace: kube-system
data:
  forward.override: |
        forward . /etc/resolv.conf
```

Or should it be like:

```
  forward.override: |
        {
            forward . /etc/resolv.conf
        }
```

If I can 'add' it to the configuration like this, does the order matter ?
Will it override it, or just add it to the configuration ?

In other words, would a configuration looking like the following be a problem ?
Would that even work ?

```
.:53 {
    errors
    health
    kubernetes cluster.local in-addr.arpa ip6.arpa {
      pods insecure
      upstream
      fallthrough in-addr.arpa ip6.arpa
    }
    prometheus :9153
    proxy . /etc/resolv.conf
    cache 30
    loop
    reload
    loadbalance
    import custom/*.override
    forward . /etc/resolv.conf <-----------
}
```

Please advice

**Comment by Vandersteen** [comment]
https://github.com/coredns/coredns/issues/3010

**Comment by jnoller** [comment]
The syntax should be the same for `proxy` and `forward`:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns-custom
  namespace: kube-system
data:
  test.server: | 
        forward . /etc/resolv.conf
        log
    }
```
